### PR TITLE
gauntlet: add resource tracker

### PR DESCRIPTION
**Pull request by JHPinto**
Isolate XpDrops into a separate entity that produces XpDropEvents, and
adjust the XpDropPlugin to work with the new events.
Add gauntlet creature death animations to the LootManager.
Add infoboxes to show gathered resources.

### CODE REVIEW

**Comment by f0rmatme** [comment]
@JHPinto we have a pr title checker, follow naming your prs like this / plugin name: info
example
"inventorysetups: updated interface"

**Comment by JHPinto** [comment]
> @JHPinto we have a pr title checker, follow naming your prs like this / plugin name: info
> example
> "inventorysetups: updated interface"

I'll keep that in mind

File: runelite-client/src/main/java/net/runelite/client/plugins/gauntlet/GauntletConfig.java [hunk 1]
```diff
@@ -321,4 +321,16 @@ default boolean displayTimerChat()
 	{
 		return false;
 	}
+
+	@ConfigItem(
+			position = 21,
+			keyName = "displayResources",
+			name = "Show raw resources gathered",
+			description = "Displays how much of each resource you have gathered.",
+			titleSection = "resources"
```
Comment (Owain94) [inline runelite-client/src/main/java/net/runelite/client/plugins/gauntlet/GauntletConfig.java]: To much tabs here (1 tab)

File: runelite-client/src/main/java/net/runelite/client/game/LootManager.java [hunk 2]
```diff
@@ -61,9 +61,65 @@
 @Slf4j
 public class LootManager
 {
-	private static final Map<Integer, Integer> NPC_DEATH_ANIMATIONS = ImmutableMap.of(
-		NpcID.CAVE_KRAKEN, AnimationID.CAVE_KRAKEN_DEATH
-	);
+	private static final Map<Integer, Integer> NPC_DEATH_ANIMATIONS = ImmutableMap.<Integer, Integer>builder()
+		.put(
+			NpcID.CAVE_KRAKEN, AnimationID.CAVE_KRAKEN_DEATH
+		)
```
Comment (Owain94) [inline runelite-client/src/main/java/net/runelite/client/game/LootManager.java]: Can you make each put single line?

```java
.put(NpcID.CAVE_KRAKEN, AnimationID.CAVE_KRAKEN_DEATH)
```

**Comment by Owain94** [comment]
Except those two small things, LGTM, thanks for the contribution!

**Pull request merged**